### DISCUSSION TITLE
[Feature] Additional photos details (useful for troubleshooting)

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [immich-app/immich#4940] [Feature Request]
### The feature

To assist in troubleshooting, I'd like the following information added to the photo details section:

- Library Name
- File path

There could be an option per user on whether to show these details (disabled by default).

This information is useful when working through issues like:
- why is this image/video not displaying properly (corrupted, not an image/photo, etc)?
- where did this image/video come from?

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by gitterspec** [comment]
I missed this and accidentally made a duplicate about showing the Source (https://github.com/immich-app/immich/discussions/13137).

If it did not come from an external library, it could just list "Uploads" instead of the external library name as the Source.

**Comment by photoserver-pro** [comment]
I’d like to add another perspective in support of this request, from the point of view of large, consolidated archives.

Showing the External Library name and source path in the asset info view is not only useful for troubleshooting, but also for day-to-day orientation and navigation once multiple external libraries are in use.

In setups where several external libraries are consolidated into a single Immich instance, assets often originate from very different logical sources (e.g. different users, devices, or historical archives). When browsing, tagging, or reviewing photos, it becomes important to immediately understand where an asset comes from, without switching context or navigating back to the External Libraries view.

This information is particularly helpful when:

- working with very large archives
- validating or curating metadata (tags, people, ratings)
- investigating duplicates or inconsistencies
- understanding the provenance of older or migrated photos

In these scenarios, the external library effectively becomes part of the asset’s identity. Making this visible in the info panel greatly improves transparency and reduces cognitive overhead when managing large collections.

I also like the idea of making this information optional or hidden by default, as mentioned above, so it remains available for power users without cluttering the UI for casual use.

Overall, this would be a small addition with a high practical value for users managing multi-source or long-term photo archives.

**Comment by timonrieger** [comment]
> [!NOTE]
> External library features are currently under feature freeze, not sure if this is within scope then.

The file path is already displayed in the UI after clicking on the info icon
<img width="391" height="145" alt="image" src="https://github.com/user-attachments/assets/5992afd6-3618-4733-9bd4-aeed7fed403f" />
Could you come up with a UI mockup? Where and what info exactly do you want to see? 
The current basis is the following:
source path is hidden by default, but accessible. The external library reference is not, I'd imagine we could put it alongside the "Appears in album" list
<img width="397" height="584" alt="image" src="https://github.com/user-attachments/assets/f2d2f42c-f156-4e45-900f-23a1d9e541e6" />

**Reply by greenturnedblue** [reply]
> The file path is already displayed in the UI after clicking on the info icon

Thanks Timon, something like what you have posted above would be amazing if it would also appear on external libraries as well. This is the info pane for an image on my external library: 
<img width="398" height="869" alt="image" src="https://github.com/user-attachments/assets/244825fa-54cb-4b26-b47e-359266201487" />

As you can see, the path is not visible.

**Reply by alextran1502** [reply]
<img width="388" height="300" alt="image" src="https://github.com/user-attachments/assets/7638a15f-0bd1-45eb-90ce-efabe6944521" />

You click on this icon to see the path

**Reply by greenturnedblue** [reply]
Thanks! I suppose it is a bit counter-intuitive to click on another info button identical to the one I just clicked, to get even _more_ info, but this will be very useful when reviewing suspected duplicate pictures now that I know where to find the file path. Thanks again

**Reply by timonrieger** [reply]
discussed on Discord [here](https://discord.com/channels/979116623879368755/1071165397228855327/1458716480735805542)

**Reply by timonrieger** [reply]
alternative/additional implementation: show the External Library the photo is in as if it was an album (in "Appears in")

**Comment by photoserver-pro** [comment]
Many thanks for the great ideas from greenturnedblue.

From a UX perspective, explicitly showing the path is significantly more intuitive for users, especially when navigating large or nested archives. It provides immediate context without requiring additional interaction.

If there are concerns about limited space in the info panel on mobile devices, this could be handled as an optional. That way, desktop and power users benefit from the added clarity, while the mobile UI can remain clean and compact if needed.

**Reply by timonrieger** [reply]
mobile does not show the icon and source path at all. would be desktop only
